aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandlerTest.java

```diff
@@ -0,0 +1,267 @@
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static software.amazon.codeguruprofiler.profilinggroup.RequestBuilder.makeRequest;
+import static software.amazon.codeguruprofiler.profilinggroup.RequestBuilder.makeValidRequest;
+
+public class UpdateHandlerTest {
+    @Mock
+    private AmazonWebServicesClientProxy proxy;
+
+    @Mock
+    private Logger logger;
+
+    private UpdateHandler subject;
+
+    private ResourceHandlerRequest<ResourceModel> request;
+
+    private final String profilingGroupName = "IronMan-2020";
+
+    private final String revisionId = "TestRevisionId-123";
+
+    private final GetPolicyRequest getPolicyRequest =
+        GetPolicyRequest.builder().profilingGroupName(profilingGroupName).build();
+
+    private final List<String> principals = Arrays.asList("arn:aws:iam::123456789012:role/UnitTestRole");
+
+    @BeforeEach
+    public void setup() {
+        proxy = mock(AmazonWebServicesClientProxy.class);
+        logger = mock(Logger.class);
+        subject = new UpdateHandler();
+    }
+
+    @Nested
+    class WhenNoPermissionsIsProvided {
+        @BeforeEach
+        public void setup() {
+            request =  makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName).build());
+            doReturn(GetPolicyResponse.builder().build())
+                .when(proxy).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+        }
+
+        @Test
+        public void testSuccess() {
+            final ProgressEvent<ResourceModel, CallbackContext> response
+                = subject.handleRequest(proxy, request, null, logger);
+
+            assertThat(response).isNotNull();
+            assertThat(response.getStatus()).isEqualTo(OperationStatus.SUCCESS);
+            assertThat(response.getCallbackContext()).isNull();
+            assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
+            assertThat(response.getResourceModel()).isEqualTo(request.getDesiredResourceState());
+            assertThat(response.getMessage()).isNull();
+            assertThat(response.getErrorCode()).isNull();
+        }
+
+        @Test
+        public void testCorrectOperationIsCalled() {
+            subject.handleRequest(proxy, request, null, logger);
+
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
```

Comment:
Can we make the check more exact (e.g. arguments?)? That way we kind-of use this mock to also test that the correct arguments are passed to the `getPolicyRequest`.